src/main/java/org/opensearch/security/user/AccountType.java

```diff
@@ -0,0 +1,26 @@
+package org.opensearch.security.user;
+
+public enum AccountType {
+
+    INTERNAL("internal"),
+    SERVICE("service");
+
+    private String name;
+
+    AccountType(String name){
+        this.name = name;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+    public static AccountType fromString(String name) {
+        for (AccountType b : AccountType.values()) {
+            if (b.name.equalsIgnoreCase(name)) {
+                return b;
+            }
+        }
+        return null;
+    }
```

Comment:
Its strange to get a null when attempting to resolve an enum from a value, either throw an error, or make the return type Optional<AccountType> so callers can handle the case where the value wasn't parsed explicitly

Reply:
As I changed the enum to represent the filter, i will return "ANY" instead of null